iceoryx_posh/include/iceoryx_posh/popo/rpc_header.hpp

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_RPC_HEADER_HPP
+#define IOX_POSH_POPO_RPC_HEADER_HPP
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+class RpcBaseHeader
```

Comment:
According to our coding guidelines we should have one header file per class - except for nested classes.

Reply:
Please comment on https://github.com/eclipse-iceoryx/iceoryx/pull/858#discussion_r654363494